src/codeu/chat/server/Model.java

```diff
@@ -67,6 +69,27 @@ public int compare(Time a, Time b) {
   private final Store<Time, Message> messageByTime = new Store<>(TIME_COMPARE);
   private final Store<String, Message> messageByText = new Store<>(STRING_COMPARE);
 
+  private final Store<User, HashSet<Uuid>> userInterestByUser = new Store<>(null);
+  private final Store<User, HashSet<Uuid>> convoInterestByUser = new Store<>(null);
+
+  public void addUserInterest(User user, User interested, UserInterests userInterests) {
```

Comment:
If you implement the code below you won't need to pass it in because you are either creating a `new UserInterests` or you are getting it from `userInterestsByUser`